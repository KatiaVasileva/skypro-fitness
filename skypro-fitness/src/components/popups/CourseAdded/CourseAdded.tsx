type CourseAddedProps = {
  onClick: () => void;
  isAddCoursePressed?: boolean;
  isDeleteCoursePressed?: boolean;
};

export const CourseAdded = ({
  onClick,
  isAddCoursePressed,
  isDeleteCoursePressed,
}: CourseAddedProps) => {
  return (
    <div className="w-full h-full overflow-x-hidden absolute z-10 bg-dark-gray/50 top-0 left-0">
      <div className="block w-screen min-h-screen mx-auto my-0">
        <div className="h-screen flex items-center">
          <div className="flex flex-col items-center gap-[30px] bg-white w-[360px] min-h-[200px] shadow-[0px_4px_67px_-12px_rgba(0,0,0,0.13)] mx-auto my-0 px-[40px] py-[40px] rounded-[30px] border-[0.7px] border-solid border-[#d4dbe5]">
            <img
              src="/img/logo.png"
              alt="logo_modal"
              className="block w-[220px] h-[35px]"
            />
            <div className="h-[60px] w-[278px] text-center">
              <p className="text-2xl text-black font-medium">
                {isAddCoursePressed
                  ? "Курс добавлен"
                  : isDeleteCoursePressed
                  ? "Курс удален"
                  : ""}
              </p>
            </div>
            <button className="btn-primary" onClick={onClick}>
              ОК
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};
