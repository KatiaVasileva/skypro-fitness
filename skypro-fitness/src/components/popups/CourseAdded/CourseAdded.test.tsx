/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import * as router from "react-router";
import { BrowserRouter } from "react-router-dom";
import { CourseAdded } from "./CourseAdded";
import UserProvider from "../../../context/UserContext";
import CoursesProvider from "../../../context/CoursesContext";
import WorkoutProvider from "../../../context/WorkoutContext";

const navigate = jest.fn();

describe("CourseAdded component", () => {
  beforeEach(() => {
    jest.spyOn(router, "useNavigate").mockImplementation(() => navigate);
  });

  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <UserProvider>
          <CoursesProvider>
            <WorkoutProvider>
              <CourseAdded onClick={function (): void {
                            throw new Error("Function not implemented.");
                        } } />
            </WorkoutProvider>
          </CoursesProvider>
        </UserProvider>
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
