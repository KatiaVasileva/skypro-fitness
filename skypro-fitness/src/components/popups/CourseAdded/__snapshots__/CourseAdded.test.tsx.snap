// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseAdded component renders correctly 1`] = `
<div>
  <div
    class="block w-full h-full overflow-x-hidden fixed z-10 bg-gray/50 top-0 left-0"
  >
    <div
      class="block w-screen min-h-screen mx-auto my-0"
    >
      <div
        class="h-screen flex items-center"
      >
        <div
          class="flex flex-col items-center gap-[30px] bg-white w-[360px] min-h-[200px] shadow-[0px_4px_67px_-12px_rgba(0,0,0,0.13)] mx-auto my-0 px-[40px] py-[40px] rounded-[30px] border-[0.7px] border-solid border-[#d4dbe5]"
        >
          <img
            alt="logo_modal"
            class="block w-[220px] h-[35px]"
            src="/img/logo.png"
          />
          <div
            class="h-[60px] w-[278px] text-center"
          >
            <p
              class="text-2xl text-black font-medium"
            />
          </div>
          <button
            class="btn-primary h-[52px] w-[235px]"
          >
            ОК
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
